src/openfermion/utils/_low_rank.py

```diff
@@ -296,21 +156,16 @@ def low_rank_spatial_two_body_decomposition(two_body_coefficients,
     truncation_errors = cumulative_error_sum[-1] - cumulative_error_sum
 
     # Optionally truncate rank and return.
-    if truncation_threshold is None and final_rank is None:
-        max_rank = full_rank
-    elif truncation_threshold is None:
-        max_rank = final_rank
-    elif final_rank is None:
-        max_rank = 1 + numpy.argmax(truncation_errors <= truncation_threshold)
+    if final_rank is None:
+        max_rank = 1 + numpy.argmax(
+            truncation_errors <= truncation_threshold)
     else:
-        raise ValueError(
-            'Cannot provide both final_rank and truncation_value.')
+        max_rank = final_rank
```

Comment:
I don't think this is the correct behavior. The truncation threshold should determine which eigenvalues are even considered. Then, `final_rank` determines whether things are further truncated. Really, `final_rank` should be renamed `max_rank`.